src/state/controls/Button.ts

```diff
@@ -71,4 +71,11 @@ export class Button extends Control<IButtonData> implements IButton {
     public giveInput(input: IButtonInput): Promise<void> {
         return this.sendInput(input);
     }
+
+    public update(changedData: Partial<IButtonData>): Promise<void> {
+        if (changedData.cooldown) {
```

Comment:
Still not entirely sure I see the problem being addressed with this. My impression was that the `set*` methods are high-level setters that are permitted to take, say, duration offset from the current time, while `.update()` is a lower-level method. It sounds like overriding it in this way would violate from the substitution principle.